document/FileFormat.mdk

```diff
@@ -2696,31 +2708,385 @@ The tool might provide a "related location" (&sect;[#result-relatedLocations]) w
 an `annotatedCodeLocation` object with the message "Insecure data entered the system here". 
 ~
 
-### `callee` property { #annotatedCodeLocation-callee }
+### `target`, `parameters`, and `variables` properties { #annotatedCodeLocation-target }
+
+Depending on the value of the `kind` property (&sect;[#annotatedCodeLocation-kind]) of this `annotatedCodeLocation` object,
+the `annotatedCodeLocation` object:
+
+* either should or shall not contain a property named `target` whose value is a string.
+* either may, should, or shall not contain a property named `parameters` whose value is an array of strings,
+each representing a variable or expression.
+* either may or shall not contain a property named `variables` whose value is a sub-object,
+each of whose keys is the name of a variable, and each of whose values is a string representing the value of the variable.
+
+These properties shall appear only in `annotatedCodeLocation` objects that are part of a `codeFlow` (&sect;[#codeFlow]).
+
+The format of the string value of the `target` property, the elements of the `parameters` array,
+the property names in the `variables` sub-object, and the property values in the `variables` sub-object,
+shall be consistent with the syntax of the programming language in which the code being analyzed was written.
+
+The interpretation of the `target`, `parameters`, and `variables` properties,
+and whether those properties may, should, or shall not be present,
+depends on the value of the `kind` property, as specified in Table [#annotatedCodeLocation-target].1.
+In this table, for each value of `kind`, "optional" means that the property may be present,
+"recommended" means that the property should be present, and "absent" means that the property shall be absent.
+
+In this table, a "variable name" may be any of the following, unless otherwise specified:
+
+* A simple variable name.
+* An object property reference.
+* An array element reference.
+* A reference to the current object.
+* Any combination of these.
```

Comment:
>Any combination of these [](start = 2, length = 24)

Why don't you use 'variable or expression' here? You list of examples doesn't cover the following case, for example (in which someone may choose to define a variable { "blah()" : "false"}.

if (blah())